configsuite/extension/ext.py

```diff
@@ -0,0 +1,206 @@
+"""Copyright 2018 Equinor ASA and The Netherlands Organisation for
+Applied Scientific Research TNO.
+
+Licensed under the MIT license.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the conditions stated in the LICENSE file in the project root for
+details.
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+"""
+
+import os
+import importlib
+
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives import unchanged_required
+from docutils import nodes
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+from configsuite.extension.ext_nodes import (
+    visit_children,
+    visit_container,
+    visit_content,
+    visit_header,
+    depart_children,
+    depart_container,
+    depart_content,
+    depart_header,
+    cs_children,
+    cs_container,
+    cs_content,
+    cs_header,
+)
+
+BASIC_EXAMPLE = {
+    "string": "default",
+    "integer": "0",
+    "number": "0.0",
+    "bool": "True",
+    "date": "01.01.1999",
+    "datetime": "01.01.1999",
+}
+
+
+def scb_static_path(app):
+    static_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "_static"))
+    app.config.html_static_path.append(static_path)
+
+
+# we are assuming several keywords.
+# I think we should provide proper user feedback if these are not included.
+def generate(schema, parent_section, parent_title=""):
+    title = schema.get(MK.Description, "default_title")[:10]
+    title = title.lower()
+    title = title.replace(" ", "_")
+    example = None
+
+    section = cs_container(
+        classes=["cs_container"], ids=["cs_{}_container".format(title)]
+    )
+
+    section += cs_header(classes=["cs_header"], text=parent_title)
+
+    section_content = cs_content(
+        classes=["cs_content"], ids=["cs_{}_content".format(title)]
+    )
+
+    section_content += nodes.paragraph(text=schema.get(MK.Description, ""))
+
+    for config_key in [MK.ElementValidators, MK.ContextValidators]:
```

Comment:
It isn't possible to distinguish between validators/transformers and even descriptions. Should have some sort of markdown to demonstrate what it is

Reply:
Created: https://github.com/equinor/configsuite/issues/115